### PR TITLE
Feature/#6 add nox task for generating the api

### LINKED ISSUE #6: Add Nox Task for generating the API
- We want to use nox to call the [openapi-python-client generator](https://github.com/openapi-generators/openapi-python-client)
- And use the config file which is stored in the repo root
  - should disable ruff
- The generator should generate the code to `exasol.saas.client.openapi`
- The generator seems to change import order (maybe dict and order issue)
  - To get a consistent import order, we perhaps can run isort

### PR DESCRIPTION
**Pull request by ckunki**
Closes #6

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+/exasol/saas/client/openapi/**  linguist-generated
```
Comment (kaklakariada) [inline .gitattributes:1]:
```suggestion
/exasol/saas/client/openapi/**  linguist-generated=true
```

File: doc/developer_guide/developer-guide.md
Comment (kaklakariada) [inline doc/developer_guide/developer-guide.md:1]: Rename file to developer_guide.md?

File: doc/developer_guide/developer-guide.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# SaaS API for Python &mdash; Developer Guide
+
+## Generate the API Model
+
+The model layer of this API has been generated from the OpenAPI specification
+in JSON format of the SaaS API https://cloud.exasol.com/openapi.json using
+[openapi-python-client](https://github.com/openapi-generators/openapi-python-client).
+
+See also
+[Documentation](https://docs.exasol.com/saas/administration/rest_api/rest_api.htm)
```
Comment (kaklakariada) [inline doc/developer_guide/developer-guide.md:10]:
```suggestion
[API Documentation](https://docs.exasol.com/saas/administration/rest_api/rest_api.htm)
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: .gitattributes (hunk 1 above)
Comment (ckunki) [reply .gitattributes:1]: See https://github.com/github-linguist/linguist/blob/master/docs/overrides.md#generated-code
And https://github.com/AlDanial/cloc/issues/722#issuecomment-1514892486

**Review by kaklakariada — DISMISSED** [review]

File: noxfile.py [hunk 3]
```diff
@@ -1,8 +1,25 @@
 """ noxfile """
 import nox
+from nox import Session
+from noxconfig import PROJECT_CONFIG
 
 # imports all nox task provided by the toolbox
 from exasol.toolbox.nox.tasks import check # pylint: disable=unused-import
```
Comment (tkilias) [inline noxfile.py:7]: The user guide of the toolbox suggest importing with star to really get all of them
https://exasol.github.io/python-toolbox/user_guide/getting_started.html#make-the-toolbox-task-available

File: noxfile.py [hunk 4]
```diff
@@ -1,8 +1,25 @@
...
+@nox.session(name="generate-api", python=False)
+def generate_api(session: Session):
+    """
+    Call openapi-python-client to generate the client api based on the
+    openapi specification for Exasol SaaS in JSON format, see
+    https://github.com/openapi-generators/openapi-python-client.
+    """
+    session.run(
+        "openapi-python-client",
+        "update",
```
Comment (tkilias) [inline noxfile.py:21]: Does update need a generate before, or can we always use update, also if the code doesn't exist yet?

File: noxfile.py [hunk 5]
```diff
@@ -1,8 +1,25 @@
...
+        "--url", "https://cloud.exasol.com/openapi.json",
```
Comment (tkilias) [inline noxfile.py:22]: hm, now the question, should we be able to overwrite this with API of the staging or internal definiton

**Review by tkilias — CHANGES_REQUESTED** [review]

**Review by kaklakariada — APPROVED** [review]

**Review by kaklakariada — DISMISSED** [review]

File: doc/developer_guide/developer_guide.md [hunk 6]
```diff
@@ -15,3 +15,21 @@ In order to regenerate the model layer please use the following command line:
...
+
+### Change the Source URL of API Model JSON Definition
+
+If you want to retrieve the JSON definition for the API model from a different
+source then just edit file `noxfile.py`.
+
+Candidates
+* https://cloud.exasol.com/openapi.json
+* https://cloud-staging.exasol.com/openapi.json
+* https://cloud-staging.exasol.com/internal-api.json
```
Comment (tkilias) [inline doc/developer_guide/developer_guide.md:27]: does not exist and should not be mentioned in a public repo

File: doc/developer_guide/developer_guide.md [hunk 7]
```diff
@@ -15,3 +15,21 @@ In order to regenerate the model layer please use the following command line:
...
 ```
+
+### Change the Source URL of API Model JSON Definition
+
+If you want to retrieve the JSON definition for the API model from a different
+source then just edit file `noxfile.py`.
+
+Candidates
+* https://cloud.exasol.com/openapi.json
+* https://cloud-staging.exasol.com/openapi.json
```
Comment (tkilias) [inline doc/developer_guide/developer_guide.md:26]: should not be mentioned in a public repo

File: exasol/saas/client/__init__.py [hunk 8]
```diff
@@ -0,0 +1 @@
+SAAS_HOST = "https://cloud-staging.exasol.com"
```
Comment (tkilias) [inline exasol/saas/client/__init__.py:5]: SAAS_HOST = "https://cloud.exasol.com"

**Review by tkilias — CHANGES_REQUESTED** [review]

**Review by kaklakariada — DISMISSED** [review]

**Review by tkilias — APPROVED** [review]

**Pull request merged**